src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt

```diff
@@ -172,7 +172,7 @@ private constructor(
       if (proxy.PROXY_TYPE_IS_SOCKS) {
```

Comment:
That code is not correct. If you switch proxy type to socks, and then disable proxy, `PROXY_TYPE_IS_SOCKS` is still set to `true` (I checked in debugger), but it should not be passed to agent.
Correct implementation would be:

```kt
 if (proxy.USE_HTTP_PROXY) {
    if (proxy.PROXY_TYPE_IS_SOCKS) {
        processBuilder.environment()["HTTP_PROXY"] = "socks://$proxyUrl"
    }
    else {
        processBuilder.environment()["HTTP_PROXY"] = "http://$proxyUrl"
        processBuilder.environment()["HTTPS_PROXY"] = "http://$proxyUrl"
    }
}